wsproto/connection.py

```diff
@@ -275,6 +275,11 @@ def events(self):
                 if frame.opcode is Opcode.PING:
                     assert frame.frame_finished and frame.message_finished
                     self._outgoing += self._proto.pong(frame.payload)
+                    yield PingReceived(frame.payload)
+
+                elif frame.opcode is Opcode.PONG:
+                    assert frame.frame_finished and frame.message_finished
+                    yield PongReceived(frame.payload)
```

Comment:
I think my quick stack idea might work.
But I agree that this is too complicated.

I propose to ignore the payload verification in wsproto, and clearly document that it is the users responsibility to compare the payloads (if it is important to their use case).